backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java

```diff
@@ -54,11 +55,21 @@ public S3BackupStore(final S3BackupConfig config, final S3AsyncClient client) {
   }
 
   @Override
-  public CompletableFuture<Void> save(final Backup backup) {
-    final var metadata = saveMetadata(backup);
-    final var snapshot = saveSnapshotFiles(backup);
-    final var segments = saveSegmentFiles(backup);
-    return CompletableFuture.allOf(metadata, snapshot, segments);
+  public CompletableFuture<BackupStatusCode> save(final Backup backup) {
```

Comment:
:question: If the future fails with exception, is it useful to return the `BackupStatusCode` here?

Reply:
You are right, I'll omit the status